bentoml/yatai/python_api.py

```diff
@@ -79,21 +87,59 @@ def upload_bento_service(bento_service, base_path=None, version=None):
         # Saving directory to path managed by LocalBentoRepository
         save_to_dir(bento_service, response.uri.uri)
 
-        upload_status = UploadStatus(status=UploadStatus.DONE)
-        upload_status.updated_at.GetCurrentTime()
-        update_bento_req = UpdateBentoRequest(
-            bento_name=bento_service.name,
-            bento_version=bento_service.version,
-            upload_status=upload_status,
-            service_metadata=bento_service._get_bento_service_metadata_pb(),
-        )
-        yatai.UpdateBento(update_bento_req)
+        update_yatai_status(yatai, bento_service)
 
         # Return URI to saved bento in repository storage
         return response.uri.uri
+    elif response.uri.type == BentoUri.S3:
+        with tempfile.TemporaryDirectory() as tmpdir:
+
+            save_to_dir(bento_service, tmpdir)
+
+            fileobj = io.BytesIO()
+            with tarfile.open(mode="w:gz", fileobj=fileobj) as tar:
+                tar.add(tmpdir, arcname=bento_service.name)
+            fileobj.seek(0, 0)
+
+            files = {'file': ('dummy', fileobj)}
+            http_response = requests.post(
+                response.uri.uri,
+                data=json.loads(response.uri.additional_fields),
+                files=files,
+            )
+
+            if http_response.status_code != 204:
+                raise BentoMLException(
```

Comment:
Also use UpdateUploadProgress to report upload failure? use the UploadStatus.ERROR